File: api/src/main/java/me/devsnox/planetsystem/api/player/PlanetPlayer.java
package me.devsnox.planetsystem.api.player;

import me.devsnox.planetsystem.api.location.PlanetLocation;
import me.devsnox.planetsystem.api.log.Logger;
import me.devsnox.planetsystem.api.planet.LoadedPlanet;
import org.bukkit.Location;
import org.bukkit.block.Block;
import org.bukkit.entity.Player;

import java.util.UUID;

public interface PlanetPlayer extends OfflinePlanetPlayer {

    @Override
    default UUID getUUID() {
        return getPlayer().getUniqueId();
    }

    Player getPlayer();

    @Override
    LoadedPlanet getPlanet();

    boolean isOnHisPlanet();

    boolean canBuild(Location location);

    default boolean canBuild(final Block block) {
        return canBuild(block.getLocation());
    }

    default boolean canNotBuild(final Location location) {
        return !canBuild(location);
    }

    default boolean canNotBuild(final Block block) {
        return !canBuild(block.getLocation());
    }

    PlanetLocation getLocation();

    Logger getLogger();

}

File: api/src/main/kotlin/me/devsnox/planetsystem/api/player/PlanetPlayer.kt
package me.devsnox.planetsystem.api.player

import me.devsnox.planetsystem.api.location.PlanetLocation
import me.devsnox.planetsystem.api.log.Logger
import me.devsnox.planetsystem.api.planet.LoadedPlanet
import org.bukkit.Location
import org.bukkit.block.Block
import org.bukkit.entity.Player

interface PlanetPlayer : OfflinePlanetPlayer {

	val player: Player

	override val planet: LoadedPlanet

	val isOnHisPlanet: Boolean

	val location: PlanetLocation

	val logger: Logger

	fun canBuild(location: Location): Boolean

	fun canBuild(block: Block): Boolean = canBuild(block.location)

}
